### PR TITLE
`treehouses services calibre-web` (fixes #1218)

### LINKED ISSUE #1218: add service calibre-web
https://hub.docker.com/r/linuxserver/calibre-web

### PR DESCRIPTION
**Pull request by law-lin**
Fixes #1218 

```
root@treehouses:~/cli# ./cli.sh services calibre-web install
Pulling calibre-web ... done
calibre-web installed
modify default environment variables by running 'cli.sh services calibre-web config edit'
root@treehouses:~/cli# ./cli.sh services calibre-web start
Starting calibre-web ... done
calibre-web started
root@treehouses:~/cli# ./cli.sh services calibre-web url
192.168.1.15:8090
```

All tests worked for me as well.

### CODE REVIEW

**Comment by LordJashin32** [comment]
```bash
root@treehouses:~/cli# bats ./tests/services/calibre-web.bats
 ✓  services calibre-web info
 ✓  services calibre-web install
 ✓  services calibre-web up
 ✓  services calibre-web start
 ✓  services calibre-web restart
 ✓  services available
 ✓  services running
 ✓  services running full
 ✓  services installed
 ✓  services installed full
 ✓  services ports
 ✓  services calibre-web port
 ✓  services calibre-web ps
 ✓  services calibre-web url
 ✓  services calibre-web autorun
 ✓  services calibre-web autorun true
 ✓  services calibre-web stop
 ✓  services calibre-web down
 ✓  services calibre-web icon
 ✓  services calibre-web cleanup

20 tests, 0 failures
root@treehouses:~/cli#
```
Tests work but can't seem to pull it up.

**Comment by law-lin** [comment]
It seems to work in the Docker container but not through local IP. 
image

File: services/install-calibre.sh [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    volumes:
+      - "/srv/calibre:/root/.calibre"
+    ports:
+      - 8080:8080
+      - 8081:8081
+    restart: unless-stopped
+EOF
+
+  # create .env with default values
+  cat << EOF > /srv/calibre/calibre.env
```
Comment (kjong) [inline services/install-calibre.sh:26]: Should rename `calibre.env` to just `.env`

File: services/install-calibre-web.sh [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  cat << EOF > /srv/calibre-web/.env
+PUID=1000
+PGID=1000
+TZ=Europe/London
+PASSWORD=PASSWORD
+DB_TYPE=sqlite
+DB_NAME=DB_NAME
+DB_HOSTNAME=DB_HOSTNAME
+DB_USERNAME=DB_USERNAME
+DB_PASSWORD=DB_PASSWORD
```
Comment (kjong) [inline services/install-calibre-web.sh:36]: I don't think these environment variables are used anywhere.
Could probably be removed.

File: services/install-calibre.sh [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+# add supported arm(s)
+function supported_arms {
+  echo "v7l"
+}
+
+# add port(s)
+function get_ports {
+  echo "8080"
+  echo "8081"
```
Comment (kjong) [inline services/install-calibre.sh:63]: Should change the ports to avoid port conflicts with other services.

**Comment by law-lin** [comment]
Closing this PR as Calibre doesn't seem to support armhf architecture and calibre-web depends on a Calibre database.

**Pull request closed without merging**